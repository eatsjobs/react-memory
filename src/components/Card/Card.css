.Card {
    margin: 0.2%;
    width: 24.6%;
    height: 24.6%;
    box-shadow: 0px 0px 0px #00ff0d;
}

.Card.Card-Glow {
  animation: glow .5s ease-in-out 0s 1 alternate;
  transform: scale(1, 1);
}

@keyframes glow {
  0% { box-shadow: 0px 0px 0px #00ff0d; }
  50% { box-shadow: 0px 0px 25px #00ff0d; }
  100% { box-shadow: 0px 0px 0px #00ff0d; }
}