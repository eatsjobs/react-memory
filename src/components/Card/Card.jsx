import React, { Component } from 'react';
import cn from 'classnames';
import sun from './asset/sun.jpg';
import heart from './asset/heart.jpg';
import flower from './asset/flower.jpg';
import gufo from './asset/gufo.jpg';
import ladybug from './asset/ladybug.jpg';
import umbrella from './asset/umbrella.jpg';

import './Card.css';

const mapImagetoName = {
    sun: sun,
    flower: flower,
    heart: heart,
    gufo: gufo,
    ladybug: ladybug,
    umbrella: umbrella
}

function CardFront(){
    return (
        <div className='Card-Front'>
            <img style={{ width: '100%' }} src='http://via.placeholder.com/265x265' alt='placeholder' />  
        </div>
    )
}

function CardBack(props){
    return (
        <div className='Card-Back'>
            <img style={{width: '100%'}} src={mapImagetoName[props.type]} alt='' />
        </div>
    )
}

export default class Card extends Component {
    constructor() {
        super(...arguments);
        this.onSelect = this.onSelect.bind(this);
    }

    onSelect(e) {
        this.props.onSelect(this.props.index, this.props.type);
    }

    render() {        
        const CardClasses = cn('Card', { 'Card-Glow': this.props.matched });
        return (
            <div className={CardClasses} onClick={this.onSelect}>
                {this.props.flipped ? <CardBack type={this.props.type} /> : <CardFront />}
            </div>
        );
    }
}