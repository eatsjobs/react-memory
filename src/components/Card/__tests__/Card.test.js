import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import Card from '../Card';

describe('<Card />', () => {
    it('should render correctly', () => {
        const onSelect = jest.fn();
        const wrapper = shallow(
            <Card type='sun' index={0} onSelect={onSelect} flipped={false} matched={false} />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should call callback', () => {
        const onSelect = jest.fn();
        const wrapper = shallow(
            <Card type='sun' index={0} onSelect={onSelect} flipped={false} matched={false} />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.simulate('click');
        expect(onSelect.mock.calls.length).toEqual(1);
        expect(onSelect).toBeCalledWith(0, 'sun');
    });
})