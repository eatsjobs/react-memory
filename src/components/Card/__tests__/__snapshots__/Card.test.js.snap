// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should call the callback with the right arguments 1`] = `
<div
  className="Card"
  onClick={[Function]}
>
  <CardFront />
</div>
`;

exports[`<Card /> should render correctly 1`] = `
<div
  className="Card"
  onClick={[Function]}
>
  <CardFront />
</div>
`;
